numalogic/tools/data.py

```diff
@@ -70,21 +118,38 @@ def __init__(
         self.val_dataset = None
 
     def setup(self, stage: str) -> None:
+        """
+        Sets up the data module by initializing the train and validation datasets.
+        """
         if stage == "fit":
             self.train_dataset = StreamingDataset(self.train_data, self.seq_len)
             if self.val_data is None:
                 return
             self.val_dataset = StreamingDataset(self.val_data, self.seq_len)
 
     def train_dataloader(self) -> TRAIN_DATALOADERS:
+        """
+        Creates and returns a DataLoader for the training dataset.
+        """
         return DataLoader(self.train_dataset, batch_size=self.batch_size)
 
-    def val_dataloader(self) -> EVAL_DATALOADERS:
+    def val_dataloader(self) -> Optional[EVAL_DATALOADERS]:
+        r"""
+        Creates and returns a DataLoader for the validation dataset if validation data is provided.
+        """
         if self.val_data is None:
-            raise DataModuleError("Validation data is not provided!")
+            _LOGGER.warning("Validation data not provided")
```

Comment:
Will this log statement carry additional context for us to determine what this warning is about (on which model, what keys, etc)?

Reply:
Same as before. We don't have keys info in models and trainers.